File: sales_commission.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sales_Commission
{


    class Employee
    {
        public int Sales { get; set; }
        public string Name { get; set; }
        public int ID { get; set; }
        public double Commission { get; set; }

        public Employee(int sales, string name, int id)
        {
            Sales = sales;
            Name = name;
            ID = id;
            Commission = GetCommission();

        }

        private double GetCommission() => Commission = Sales * 500;
        
    }

    class Program
    {
        static int IntValidation(string inp)
        {
            int temp;
            while (true)
            {
                Console.Write(inp);
                if (int.TryParse(Console.ReadLine(), out temp))
                    break;

                Console.Clear();
                Console.WriteLine("[x] Ensure your input is a number!");
            }

            return temp;
        }
        static void Main(string[] args)
        {
            int employee_count;
            int _sales;
            int _id;

            employee_count = IntValidation("Input Employee counter : ");
            Employee[] employeeArray = new Employee[employee_count];

            for (int i = 0; i < employee_count; i++)
            {
                Console.Write($"Employee {i + 1} Name : ");
                var _name = Console.ReadLine();

                _sales = IntValidation($"Employee {i + 1} Sales : ");
                _id = IntValidation($"Employee {i + 1} ID : ");

                employeeArray[i] = new Employee(_sales, _name, _id);
                Console.Clear();
            }

            Array.Sort(employeeArray, (x, y) => x.Sales.CompareTo(y.Sales));
            Array.Reverse(employeeArray);
            // Sort Employee Array by Sales Property
            // Reverse array to order by descension (most sales -> least sales) 


            double bonus = employeeArray[0].Commission *.15;
            employeeArray[0].Commission += bonus;
            // Give most sales a 15% bonus

            Console.WriteLine($"{employeeArray[0].Name} [ID - {employeeArray[0].ID}] earned a bonus 15% (£{bonus:N2}) for having the most sales!\n");
            // Output top sale Employee + Output bonus earned

            foreach (Employee employee in employeeArray)
            {
                Console.WriteLine(
                    $"Employee" +
                    $"    Name - {employee.Name}" +
                    $"    ID - {employee.ID}" +
                    $"    Sales - {employee.Sales}" +
                    $"    Commission - £{employee.Commission:N2}"
                );
            }
            // Output all employee information (Name; ID; Sales; Commission)

            Console.WriteLine("--------------------------");
            Console.WriteLine($"Total Properties Sold - {employeeArray.Sum(employee => employee.Sales)}");
            // Output all properties sold in the week

            Console.WriteLine($"Total Sales Commission - £{employeeArray.Sum(employee => employee.Commission):N2}" );
            // Output total sales commission for the week

            Console.ReadKey();
        }
    }
}

File: sales_commission.py
from os import system, name

class Employee:
    def __init__(self, sales : int, name, id : int):
        self.sales = sales
        self.name = name
        self.id = id
        self.commission = self.sales * 500

    @property
    def info(self):
        return (
            f"Employee [{self.id}]: \n" +
            f"    Name - {self.name} \n" + 
            f"    ID - {self.id} \n" +  
            f"    Sales - {self.sales} \n" + 
            f"    Commission - £{self.commission:,.2f} \n"
        )
      
if "__main__" == __name__:
    employee_count = int(input("Input Employee Counter : "))
    employee_arr = []

    for i in range(employee_count):
        while True:
            try:
                _name = input(f"Employee {i + 1} Name : ").capitalize()
                _sales = int(input(f"Employee {i + 1} Sales : "))
                _id = int(input(f"Employee {i + 1} ID : "))

                system('cls' if name == 'nt' else 'clear')
                employee_arr.append(Employee(_sales, _name, _id))
                break

            except ValueError:
                print("[x] Ensure your input fields are valid.")
        
    employee_arr.sort(key=lambda x: x.sales, reverse=True)

    bonus = employee_arr[0].commission * .15
    employee_arr[0].commission += bonus
    print(f"{employee_arr[0].name} [ID - {employee_arr[0].id}] earned a bonus 15% (£{bonus:.2f}) for having the most sales!\n")

    for index, employee in enumerate(employee_arr):
        print(employee.info)

    print("-----------------")
    print(f"Total Properties Sold: - {sum(x.sales for x in employee_arr)}")
    print(f"Total Sales Commission: - £{sum(x.commission for x in employee_arr):,.2f}")
